client/src/main/java/org/apache/iotdb/client/AbstractClient.java

```diff
@@ -805,34 +503,100 @@ private static void importCmd(String specialCmd, String cmd, IoTDBConnection con
     } catch (SQLException e) {
       println(String.format("Failed to import from %s because %s",
           cmd.split(" ")[1], e.getMessage()));
-      handleException(e);
     } catch (TException e) {
       println("Cannot connect to server");
-      handleException(e);
     }
   }
 
   private static void executeQuery(IoTDBConnection connection, String cmd) {
     long startTime = System.currentTimeMillis();
-    try (Statement statement = connection.createStatement();) {
+    try (Statement statement = connection.createStatement()) {
       ZoneId zoneId = ZoneId.of(connection.getTimeZone());
       statement.setFetchSize(fetchSize);
       boolean hasResultSet = statement.execute(cmd.trim());
       if (hasResultSet) {
-        ResultSet resultSet = statement.getResultSet();
-        output(resultSet, printToConsole, zoneId);
-        if (resultSet != null) {
-          resultSet.close();
+        try (ResultSet resultSet = statement.getResultSet()) {
+          ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+          int columnLength = resultSetMetaData.getColumnCount();
+          List<Integer> maxSizeList = new ArrayList<>(columnLength);
+          List<List<String>> lists = cacheResult(resultSet, maxSizeList, columnLength,
+              resultSetMetaData, zoneId);
+          output(lists, maxSizeList);
         }
       }
     } catch (Exception e) {
       println("Msg: " + e.getMessage());
```

Comment:
There is a null pointer exception, I can not print result in my client. Try to fix it...

Reply:
fixed.